### PR TITLE
Adding orders to process method and test

### PR DESCRIPTION
**Pull request by eapg**
* Adding orders to process method
* Adding order repository test for orders to process method
* Adding order controller unit test for orders to process method
* Adding order controller integration test for orders to process method
* Adding enum class for order status

Orders to process method:

This method will extract all the orders of the order repository and will terminate by the order status wich of the orders are
eligible to be process. in this case the method will filter and return the orders with status "new_order"

### CODE REVIEW

File: src/constants/order_status.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+from enum import Enum
+
+
+class OrderStatus(Enum):
+    NEW_ORDER = "new_order"
```
Comment (reynoldmorel) [inline src/constants/order_status.py:5]: Could briefly explain the lifecycle of an order?

File: src/tests/api/controllers/order_controller_integration_test.py [hunk 2]
```diff
@@ -110,3 +111,17 @@ def test_update_order_from_repository_using_controller(self):
...
+        order_2 = build_order()
+        order_3 = build_order(order_status=OrderStatus.COMPLETED)
+
+        self.order_controller.add(order_1)
+        self.order_controller.add(order_2)
+        self.order_controller.add(order_3)
+
+        orders_to_process = self.order_controller.get_orders_to_process()
+
+        self.assertEqual(orders_to_process, [order_1, order_2])
```
Comment (reynoldmorel) [inline src/tests/api/controllers/order_controller_integration_test.py:127]: Could you assert the call in the repository layer?

File: src/lib/repositories/impl/order_repository_impl.py [hunk 3]
```diff
@@ -25,3 +26,11 @@ def delete_by_id(self, order_id):
     def update_by_id(self, order_id, order):
         current_order = self.get_by_id(order_id)
         current_order.order_details = order.order_details or current_order.order_details
+
+    def get_orders_to_process(self):
+        orders = self.get_all()
+
+        orders_to_process = [
```
Comment (reynoldmorel) [inline src/lib/repositories/impl/order_repository_impl.py:33]: I think there are built-in functions in python to apply filters in lists: https://www.geeksforgeeks.org/filter-in-python/

**Review by reynoldmorel — CHANGES_REQUESTED** [review]

File: src/tests/api/controllers/order_controller_integration_test.py (hunk 2 above)
Comment (eapg) [reply src/tests/api/controllers/order_controller_integration_test.py:127]: done.

File: src/constants/order_status.py (hunk 1 above)
Comment (eapg) [reply src/constants/order_status.py:5]: The lifecycle will be:

when an order is create its first status is **"new_order"**, after the order passed through all the checks, like ingredient availability validation,if the order don't passed any of the checks the order will pass to **"cancelled"** status, the order is passed to **"placed_order"** status, after that an available chef take the order and change its status to **"in_process"**. when the order runs out of preparation time the chef put the order in **"complete**" status.

File: src/lib/repositories/impl/order_repository_impl.py (hunk 3 above)
Comment (eapg) [reply src/lib/repositories/impl/order_repository_impl.py]: done. i used filter() + lambda function.

File: src/constants/order_status.py (hunk 1 above)
Comment (reynoldmorel) [reply src/constants/order_status.py:5]: Nice thanks, but could you write this in the code?

**Review by reynoldmorel — APPROVED** [review]

**Pull request merged**